### DISCUSSION TITLE
Unreliable network (Sonoff Dongle-E)

### DISCUSSION THREAD

**Discussion by ra-kal** [Koenkk/zigbee2mqtt#19763] [General]
OK, my Zigbee network is not reliable and I need some help/ideas.

The connected devices usually work, but...  Sometimes switching lights from HA is noticably delayed or does not work at all. The same happens in the opposite direction, where state of a switch operated physically is not updated in HA. This happens even with devices that are ~2-3m away from the Sonoff Dongle-E coordinator.

I tried to sumarize my setup and what I have already tried to fix it, so it is rather longer text. Therefore, I directly start with the questions:

- Can Zigbee network reliably work when there are multiple WiFi routers in surronding appartments?
- Is the Sonoff Dongle-E not working reliably? Eg. messages are sometimes not sent/received?
- How to diagnose Zigbee network? Is there any possibility to see the error rate from the coordinator?
- Besides the original 6.10.3 firmware, is there any other recommended FW version to switch to?
- Should I trash the Dongle-E and buy different coordinator?

The long story...

I have already tried to fix it, but nothing solved it.
1) move NUC/Coordinator
2) add 1.5m USB cable
3) add USB hub between NUC and coordinator
4) change Zigbee channel 11 -> 25

HW:
- Intel NUC
- Sonoff Dongle-E connected via HUB and 1.5m cable. NUC only provides USB 3.x ports.

SW:
- Proxmox
- Ubuntu server
- HA supervised
- Zigbee2Mqqt

Other wireless networks
- WiFi: There are many WiFi networks around so it is not possible to find any free channel. My two WiFi routers however use channels that not interfere with Zigbee. WiFi in the NUC is not used.
- BT: The bluetooth on the NUC is enabled and used to receive data from BT temperature sensors. 
 
Zigbee network:
- 13 powered devices
- 13 battery devices
The devices are within 10m apart from coordinator. LQI is usually 120+, but there are some devices, which are like 3, away, but show LQI around 100 or below. In the map view the signal strength of devices directly connected to the coordinator is usually 40/120 - 60/160.

I'm out of ideas :-( I have one IKEA button directly linked with LED strip. This works all the time. But the same button presses are sometimes not visible in HA. The same happens with the LED strip, which sometimes does not react to commands from HA. 

My assumption therefore is that the Sonoff Dongle-E coordinator could be the culprit. But the recommended RF interference optimizations didn't helped.

A typical step would be to update the FW, but here the infos are not clear. They only usually states, that the "old" 6.10.3 version is stable and works fine. But in another chat has somebody written, that he experiences bad signal quality with this version in comparison to the newer versions.

**Comment by Interloping** [comment]
I think the Dongle-E is the issue. I recently switched from a Conbeee II to a Dongle-E. It's been a string of headaches for me. The Dongle-E regularly crashes zigbee2mqtt and this means a long period while it restarts where nothing is responsive. It is listed as being only experimentally supported. Unfortunately the website I bought it from listed it as supported zigbee2mqtt, without mentioning that's not quite the case. Lesson learned.

**Reply by ra-kal** [reply]
I do not observe any crashes. Only soporadic delayed or lost messages.

**Comment by ra-kal** [comment]
I have one powered device about 3m away from the coordinator. It does not pair directly to coordinator, but only through another router. After a while however it selects direct connection to the coordinator and the LQI dropps below 100 or lower. Another device sometimes shows LQI 170 and sometimes 60. 

OK, I think I need to try another coordinator.
Is the Skyconnect any better? Would you recommend it?

The SkyConnect is based on the same EZSP stack, It is a question, if the problem in my case is the stack or the HW.

**Reply by lux73** [reply]
buy any CC2652x based Coordinator

**Reply by Ra72xx** [reply]
There seems to be some instability with this sort of coordinators since 1.33.2:
https://github.com/Koenkk/zigbee2mqtt/issues/19747

**Reply by ra-kal** [reply]
> There seems to be some instability with this sort of coordinators since 1.33.2: #19747
Thanks for the info.

It looks related, but the behavior is not exactly the same. I had the connection issues from the beginning, e.g. also with older Z2M versions. I have no "stuck" devices that would not be controllable at all. They just do not update or react to changes for a while. For example if I turn on a light by a physical button, the update message from the device will not reach Z2M and therefore in HA it still looks like the light is off. But it usually synchronizes again on the next change either from the device or from HA. So in my case the devices are not stuck for ever, but just sometimes the messages are lost or have noticable delay.

Interestingly I have two same smart plugs TuYa TS011F_plug_1, they are like 10cm apart from each other, and 3m from coordinator. One plug has LQI 84 (in map 84/12) and the other 164 (in map 164/69).

It would be interesting to know, if it is caused by the devices (mostly Tuya, but Lidl also), EZSP stack, the coordinator HW (antenna), or incompatibility with Z2M.

Unfortunately the CC2652x based adapters are sold out in my country and the sellers are not going to stock up them again in favor for the new coordinators based on EFR chip.

**Comment by ra-kal** [comment]
OK, so I finally bought Sonoff Dongle-P with CC2652P chip. I updated it to the latest Koenkk FW 3.x.0 V20230507 and migrated the configuration from the Dongle-E (even thought it seemd not to work at first).

But the network is much worse now :-( The reported LQIs dropped from 160+ to 30-50 max for all devices. The commands from HA have very noticable delay, and in many cases it does not work at all. 

What is happening here? Is the praised CC2652P so bad in signal quality? 
I have already tried to set transmission power to 9, but it does not seem to have any effect.

**Reply by charlesomer** [reply]
This sounds similar to issues I've been experiencing too: https://github.com/Koenkk/zigbee2mqtt/issues/19416 ?

**Comment by ra-kal** [comment]
Finally I decided to switch to Sonoff Dongle-P and re-pair all devices. I let the network stabilize overnight. The LQIs compared to Dongle-E are much lower in range 30-50, but some devices have even lower number. Sometimes I see devices reporting even 0-10.

Result: The network still very unreliable with the same manifestation of errors, eg. wrong state in HA or failures to control devices.

Another behavior is that while updating the map view I get "Failed to execute LQI for ..." errors or ping errors.

I'm out of ideas. How to diagnose the network and what is causing the problems? Are some of my devices causing it? Or is the Zigbee really that bad in coexisting in 2.4GHz frequency with Wifi and Bluetooth networks and dealing with network errors? E.g. if there are wifi routers around, that I have no control of, spread in all channels, is zigbee not usable at all.
